manager-bundle/src/Composer/ScriptHandler.php

```diff
@@ -18,107 +18,74 @@
-
-        $event->getIO()->write('<info>Done! Please open the Contao install tool or run contao:migrate on the command line to make sure the database is up-to-date.</info>');
-    }
-
-    public static function purgeCacheFolder(): void
-    {
-        $filesystem = new Filesystem();
-        $filesystem->removeDirectory(Path::join(getcwd(), 'var/cache/prod'));
-    }
-
-    /**
-     * Adds the app directory if it does not exist.
-     */
-    public static function addAppDirectory(): void
-    {
-        $filesystem = new Filesystem();
-        $filesystem->ensureDirectoryExists(Path::join(getcwd(), 'app'));
-    }
+        trigger_deprecation('contao/manager-bundle', '4.11', 'Using ScriptHandler::initializeApplication has been deprecated and will be removed in Contao 5.0. Use the InitializeApplicationCommand instead.');
+
+        $command = array_filter([
+            'contao-console',
+            'contao:initialize-application',
+            $event->getIO()->isDecorated() ? '--ansi' : '',
+        ]);
+
+        $event->getIO()->write(
+            sprintf(
+                '<warning>Please edit your root composer.json and set "%s" to "%s" instead of calling "ScriptHandler::initializeApplication" directly.</warning>',
+                $event->getName(),
+                implode(' ', $command)
+            )
+        );
 
-    /**
-     * @throws \RuntimeException
-     */
-    private static function executeCommand(array $cmd, Event $event, string $env = 'prod'): void
-    {
-        $phpFinder = new PhpExecutableFinder();
+        // Purge the cache, to make sure the 'contao:initialize-application' command is found
+        self::purgeCacheFolder();
 
-        if (false === ($phpPath = $phpFinder->find())) {
+        if (false === ($phpPath = (new PhpExecutableFinder())->find())) {
             throw new \RuntimeException('The php executable could not be found.');
         }
 
-        $command = array_merge(
-            [$phpPath, __DIR__.'/../../bin/contao-console'],
-            $cmd,
-            ['--env='.$env, $event->getIO()->isDecorated() ? '--ansi' : '--no-ansi']
-        );
-
-        if ($verbose = self::getVerbosityFlag($event)) {
-            $command[] = $verbose;
-        }
+        $command[0] = Path::join(__DIR__.'/../../bin', $command[0]);
+        array_unshift($command, $phpPath);
 
         // Backwards compatibility with symfony/process <3.3 (see #1964)
         if (method_exists(Process::class, 'setCommandline')) {
             $command = implode(' ', array_map('escapeshellarg', $command));
         }
 
         $process = new Process($command);
-
-        // Increase the timeout according to terminal42/background-process (see #54)
-        $process->setTimeout(500);
+        $process->setTimeout(null);
 
         $process->run(
             static function (string $type, string $buffer) use ($event): void {
                 $event->getIO()->write($buffer, false);
             }
         );
-
```

Comment:
Removing this means the `ScriptHandler` will continue to run the next command even though one failed. I don't think that is intended?

Reply:
There is only command executed here (`contao:initialize-application`).